osu.Game.Rulesets.Mania/Beatmaps/Patterns/Legacy/DistanceObjectPatternGenerator.cs

```diff
@@ -76,7 +83,7 @@ public override IEnumerable<Pattern> Generate()
 
             foreach (var obj in originalPattern.HitObjects)
             {
-                if (!Precision.AlmostEquals(EndTime, obj.GetEndTime()))
+                if (EndTime != (int)Math.Round(obj.GetEndTime()))
```

Comment:
Could this be `Math.Floor` for consistency with the ctor? Not a huge deal really but I'm just semi-curious if there's a reason for why the two differ.

Reply:
The calculation in the ctor is directly from stable. The calculation here is just making sure that lazer's storage as `double` doesn't lead to incorrect comparisons. It passes the tests but I don't believe this can be replaced with `Math.Floor` - I'll run across my full suite of beatmaps.

Reply:
If it's verbatim from stable then not worth. Maybe leave a comment to not touch at best.

Reply:
Changing this one to a `Math.Floor()` doesn't affect any beatmaps I have but I'm still not sure how I feel about that in principle.

Reply:
Nah it's fine. As I said I was mostly curious if there was reason for the difference other than stable gospel.